Sourcery/Models/TypeName.swift

```diff
@@ -59,13 +67,120 @@ final class TypeName: NSObject, AutoDiffable {
         }
     }
 
-    /// sourcery: skipEquality
-    /// sourcery: skipDescription
+    // sourcery: skipEquality
     var isVoid: Bool {
         return name == "Void" || name == "()"
     }
 
+    var isTuple: Bool {
+        return tuple != nil
+    }
+
+    private var _tuple: TupleType?
+    var tuple: TupleType? {
+        if _tuple == nil { _tuple = TupleType(self.actualTypeName?.name ?? self.name) }
+        return _tuple
+    }
+
     override var description: String {
+        if let actualTypeName = actualTypeName {
+            return "\(name) aka \(actualTypeName.name)"
+        }
         return name
     }
 }
+
+final class TupleType: NSObject, AutoDiffable {
+    let name: String
+
+    final class Element: NSObject, AutoDiffable, Typed {
+        let name: String
+        let typeName: TypeName
+
+        // sourcery: skipEquality, skipDescription
+        var type: Type?
+
+        init(name: String, typeName: String, type: Type? = nil) {
+            self.name = name
+            self.typeName = TypeName(typeName)
+            self.type = type
+        }
+    }
+
+    lazy private(set) var elements: [Element] = {
+        let trimmedBracketsName = String(self.name.characters.dropFirst().dropLast())
+        return trimmedBracketsName
```

Comment:
I really think Parsing should happen in the `FileParser` or one of associated files rather than model objects